src/directives/index.ts

```diff
@@ -96,11 +142,20 @@ const numbering = (directives: Directives): RuleCore => (state) => {
   const { tokens } = state;
   for (let index = 0; index < tokens.length; index += 1) {
     const token = tokens[index];
-    if (token.type === 'container_directives_open') {
+    if (token.type === DirectiveTokens.open) {
       const directive = getDirective(directives, token.attrGet('kind'));
       if (directive?.numbered) {
-        const { name } = token.meta?.opts;
-        const target = newTarget(state, name, directive.numbered);
+        const { name, label } = token.meta?.opts;
+        const target = newTarget(state, name || label, directive.numbered);
+        token.meta.target = target;
+      }
+    }
+    if (token.type === DirectiveTokens.fence) {
+      const directive = getDirective(directives, token.attrGet('kind'));
+      const { name, label } = token.meta?.opts;
+      // Only number things if the directive supports numbering AND a name or label is provided
+      if (directive?.numbered && (name || label)) {
+        const target = newTarget(state, name || label, directive.numbered);
```

Comment:
@choldgraf I am using `name` or `label` for things like equations a bit interchangably. The RST docs seem to say `name` is common, but EB suggests `label` for things like math? Let me know what the convention is!! This seems to work for now.

Reply:
Hmmm - my personal understanding is that label is the "thing you reference" elsewhere (e.g. `myeq1`). E.g., the [rst quickstart](https://docutils.sourceforge.io/docs/user/rst/quickref.html) uses `label` quite a lot, very little "name". Where do you see `name` recommended in rst?

Reply:
This is where I was going to:
https://docutils.sourceforge.io/docs/ref/rst/directives.html#math

Reply:
The other place I have been looking is here:
https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#math

Which does say `label` -- missed that before.

Reply:
![image](https://user-images.githubusercontent.com/913249/99916378-39840700-2cc7-11eb-84a3-fa09ba93e7d1.png)

Reply:
Upon further conversation:

* Use name. Label is directive specific (e.g. equation).

Reply:
See also https://github.com/sphinx-doc/sphinx/issues/8476